### PR TITLE
[RESOLVE]#858 Rework passive assignment logic

### PR DESCRIPTION
**Pull request by DannyDelic**
- Renamed variable to a more appropriate name
- Now finds name by checking the attached Passive.cs file in a champion folder
- Parses the *.cs-file as text and uses a regex match, to find the passiveName
- Should be somewhat resilliant towards similar-sounding comments, as it checks for the following interface in the pattern

### CODE REVIEW

File: GameServerLib/Content/CharData.cs [hunk 1]
```diff
@@ -10,8 +11,22 @@ namespace LeagueSandbox.GameServer.Content
...
     {
         public string PassiveNameStr { get; set; } = "";
-        public string PassiveLuaName { get; set; } = "";
+        public string PassiveAbilityName { get; set; } = "";
         public int[] PassiveLevels { get; set; } = { -1, -1, -1, -1, -1, -1 };
+
+        //TODO: Extend into handling several passives, when we decide on a format for that case.
+        public static string GetPassiveAbilityNameFromScriptFile(string champName)
+        {
+            var inputPassiveFile = File.ReadAllText($"..\\..\\..\\Content\\LeagueSandbox-Scripts\\Champions\\{champName}\\Passive.cs");
```
Comment (abosma) [inline GameServerLib/Content/CharData.cs]: You would rather not hard-code this to LeagueSandbox-Scripts.

Comment (DannyDelic) [reply GameServerLib/Content/CharData.cs]: Is there any environment-variable I can point at in this case then?

Comment (abosma) [reply GameServerLib/Content/CharData.cs]: Game has the config file for the environment variables like content location, which I believe CharData contains.

Comment (lzardy) [reply GameServerLib/Content/CharData.cs]: Check Package.cs as well if you haven't already.

Comment (DannyDelic) [reply GameServerLib/Content/CharData.cs]: There is an attached _contentManager in the CharData class, but the variables for the PackagePath and Package is empty - so I have been forced to alter the ICharData interface to get the desired strings, which you can see in my next commit.

Comment (abosma) [reply GameServerLib/Content/CharData.cs]: Using _contentManager.GetAllLoadedPackages() gives me all packages loaded together with their names and paths, here's and example picture: [Code example](https://i.imgur.com/vAudcSW.png)

Comment (DannyDelic) [reply GameServerLib/Content/CharData.cs]: Didn't load the packages, thanks :)

Comment (DannyDelic) [reply GameServerLib/Content/CharData.cs]: Should be resolved in the latest commit now.

**Comment by DannyDelic** [comment]
> You should use an IPackage list instead of Package since there is probably going to be an update for different kinds of packages.

Noted and implemented.

File: GameServerLib/Content/CharData.cs [hunk 2]
```diff
@@ -10,8 +13,32 @@ namespace LeagueSandbox.GameServer.Content
...
+                if (File.Exists(path))
+                {
+                    var inputPassiveFile = File.ReadAllText(path);
+
+                    string pattern = @"class (?<passiveName>\w+) : IGameScript";
+
+                    RegexOptions options = RegexOptions.Multiline;
+
+                    var passiveName = Regex.Match(inputPassiveFile, pattern, options).Groups["passiveName"].Value;
+
```
Comment (danil179) [inline GameServerLib/Content/CharData.cs:31]: Remove blank lines.

File: GameServerLib/Content/CharData.cs [hunk 3]
```diff
@@ -10,8 +13,32 @@ namespace LeagueSandbox.GameServer.Content
...
+        {
+            foreach (var package in packages)
+            {
+                var path = $"{package.PackagePath}\\Champions\\{champName}\\Passive.cs";
+
+                if (File.Exists(path))
+                {
+                    var inputPassiveFile = File.ReadAllText(path);
+
+                    string pattern = @"class (?<passiveName>\w+) : IGameScript";
```
Comment (danil179) [inline GameServerLib/Content/CharData.cs:28]: We need to discuss whether we want to use a way like that (inferring names from classes. Maybe we want to do some reflection magic to have a better handling)

**Review by danil179 — CHANGES_REQUESTED** [review]
Fix the style issues and we will discuss this way of deducing the passive names.

File: GameServerLib/Content/CharData.cs (hunk 3 above)
Comment (DannyDelic) [reply GameServerLib/Content/CharData.cs:28]: I don't have much experience using the Reflections library, could you explain how you would want to use it for this use-case`?

Comment (danil179) [reply GameServerLib/Content/CharData.cs:28]: See this https://github.com/LeagueSandbox/GameServer/blob/indev/GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs (this class uses a lot of reflection).

It should be something like:
* Get the file
* Create some assembly object from it
* Get class name
* Infer passive name from it

Comment (danil179) [reply GameServerLib/Content/CharData.cs:28]: Let's delay it for another PR as compiling in real time only to find the passive name is not efficient at all. (We need to add functions to the `CSharpScriptEngine` to allow that)

**Review by danil179 — APPROVED** [review]
Approved. Issue will be added regrading the problematic regex (I can see how one can misuse this to break abilities/ crash the game).

**Pull request merged**